### PR TITLE
OCPQE-19123: Add LVMS Operator CI test jobs

### PR DESCRIPTION
**Pull request by Phaow**
Add LVMS CI tests

### CODE REVIEW

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-compact-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-sno-ovn-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f7

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f14

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-baremetal-ipi-ovn-lvms-f14

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f14

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-lvms-f14

File: ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/ipi/ovn/lvms/cucushift-installer-rehearse-baremetalds-ipi-ovn-lvms-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    - chain: cucushift-installer-rehearse-baremetalds-ipi-provision
+    - chain: storage-conf-csi-optional-topolvm
+    post:
+    - chain: cucushift-installer-rehearse-baremetalds-ipi-deprovision
+    env:
+      DEVSCRIPTS_CONFIG: |
+        IP_STACK=v4
+        NETWORK_TYPE=OVNKubernetes
+        VM_EXTRADISKS=true
+        VM_EXTRADISKS_LIST="vda vdb"
```
Comment (radeore) [inline ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/ipi/ovn/lvms/cucushift-installer-rehearse-baremetalds-ipi-ovn-lvms-workflow.yaml:14]: Just trying to understand, here we are not specifically setting `LVM_CLUSTER_DEVICE_PATH`?

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/sno/lvms/cucushift-installer-rehearse-aws-ipi-sno-lvms-workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+workflow:
+  as: cucushift-installer-rehearse-aws-ipi-sno-lvms
+  steps: 
+    pre:
+    - chain: cucushift-installer-rehearse-aws-ipi-sno-provision
```
Comment (radeore) [inline ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/sno/lvms/cucushift-installer-rehearse-aws-ipi-sno-lvms-workflow.yaml:5]: General question, for LVMS on SNO we need  vm_instance with 8 vCpu, I usually overwrite vm_type: 'm6i.2xlarge' to setup OCP AWS flexy cluster, do we have this vm_type setup in this chain?

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/sno/lvms/cucushift-installer-rehearse-aws-ipi-sno-lvms-workflow.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    - ref: storage-create-aws-extra-disks
+    - chain: storage-conf-csi-optional-topolvm
+    post:
+    - chain: cucushift-installer-rehearse-aws-ipi-deprovision
+    env:
+      LVM_CLUSTER_DEVICE_PATH: "/dev/nvme2n1"
```
[the remark this answers is not in the document]
Comment (Phaow) [reply ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/sno/lvms/cucushift-installer-rehearse-aws-ipi-sno-lvms-workflow.yaml:11]: Hmm, for this workflow the SNO cluster have 4 disks, 2 is created when install `nvme0n1` and `nvme1n1`, we create 2 extra disks used for lvm test `nvme2n1` and `nvme3n1`.

File: ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/ipi/ovn/lvms/cucushift-installer-rehearse-baremetalds-ipi-ovn-lvms-workflow.yaml (hunk 1 above)
Comment (Phaow) [reply ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/ipi/ovn/lvms/cucushift-installer-rehearse-baremetalds-ipi-ovn-lvms-workflow.yaml:14]: Yeah, correct.

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/sno/lvms/cucushift-installer-rehearse-aws-ipi-sno-lvms-workflow.yaml (hunk 2 above)
Comment (Phaow) [reply ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/sno/lvms/cucushift-installer-rehearse-aws-ipi-sno-lvms-workflow.yaml:5]: Good point, currently in this chain we don't overwrite vm_type, I'll overwrite it in our workflow.

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetal-ipi-ovn-lvms-f14

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-aws-ipi-sno-lvms-f14

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-aws-ipi-sno-lvms-f14

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-aws-ipi-sno-lvms-f14

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-aws-ipi-sno-lvms-f14

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetal-ipi-ovn-lvms-f14

**Comment by Phaow** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetal-ipi-ovn-lvms-f14

**Comment by Phaow** [comment]
@yunjiang29 @jianlinliu @liangxia @kuiwang02 Could you please help take a look when you get a chance? Thanks!

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__arm64-nightly.yaml [hunk 4]
```diff
@@ -181,6 +181,24 @@ tests:
     test:
     - chain: openshift-e2e-test-qe
     workflow: cucushift-installer-rehearse-aws-ipi-registry-s3-cloudfront
+- as: aws-ipi-sno-lvms-f14
```
[the remark this answers is not in the document]
Comment (Phaow) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__arm64-nightly.yaml:184]: @jianlinliu Good point, yeah the LVMS Operator support multi arch, but currently it released as standlone and many cases will reconfig the storage maybe impact other subteam cases, so we add the jobs only execute LVMS cases and these 2 jobs execute both non-destructive and destructive test cases(in 30min). We use aws cover the arm64 and BM cover the amd64, it'll produce less costs than separate jobs `non-destructive` and `destructive`. WDYT?

[the remark this answers is not in the document]
Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__arm64-nightly.yaml:184]: okay, got it.

**Comment by Phaow** [comment]
Rebase latest codes to sovle the merge conflicts.

File: ci-operator/step-registry/storage/create/aws-extra-disks/storage-create-aws-extra-disks-commands.sh [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+	# shellcheck disable=SC1090
+	source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+# Create an AWS EBS volume, volInfraTag will help delete volume when deleting cluster 
+function create_ebs_volume(){
+  if [ $# -ne 6 ] ;then
+    echo "Usage: create_ebs_volume <volName> <availabilityZone> <volSize> <volType> <volInfraTag>" && return 1
```
Comment (liangxia) [inline ci-operator/step-registry/storage/create/aws-extra-disks/storage-create-aws-extra-disks-commands.sh:22]: `$#` check for 6 params, while the example show 5.  Is it expected ?

Comment (Phaow) [reply ci-operator/step-registry/storage/create/aws-extra-disks/storage-create-aws-extra-disks-commands.sh]: Good catch, yeah it should be 6 params, missed the `region`! Fixed, thanks!

**Comment by Phaow** [comment]
Hello @jsafrane, I added some refs used for LVMS Operator tests, since the refs maybe only used by storage related components I added them in our folder `ci-operator/step-registry/storage`(The rehearse results all looks good). Could you please help take a look when you get a chance for the final lgtm/approve. Thanks!

**Comment by jsafrane** [comment]
/approve
step-registry/storage changes

**Pull request merged**